### PR TITLE
Improve candid/macaroon based authentication

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

**Comment by stgraber** [comment]
Needs `make i18n`

File: test/suites/serverconfig.sh [hunk 1]
```diff
@@ -19,10 +19,10 @@ test_server_config() {
   curl --unix-socket "$LXD_DIR/unix.socket" "lxd/1.0" | jq .metadata.auth_methods | grep tls
   # only tls is enabled by default
   ! curl --unix-socket "$LXD_DIR/unix.socket" "lxd/1.0" | jq .metadata.auth_methods | grep macaroons
```
Comment (stgraber) [inline test/suites/serverconfig.sh]: s/macaroons/candid/

File: test/suites/serverconfig.sh [hunk 2]
```diff
@@ -19,10 +19,10 @@ test_server_config() {
...
-  lxc config set core.macaroon.endpoint "https://localhost:8081"
+  lxc config set core.candid.endpoint "https://localhost:8081"
   # macaroons are also enabled
   curl --unix-socket "$LXD_DIR/unix.socket" "lxd/1.0" | jq .metadata.auth_methods | grep macaroons
```
Comment (stgraber) [inline test/suites/serverconfig.sh]: s/macaroons/candid/

File: client/lxd.go [hunk 3]
```diff
@@ -365,7 +365,10 @@ func (r *ProtocolLXD) websocket(path string) (*websocket.Conn, error) {
 func (r *ProtocolLXD) setupBakeryClient() {
 	r.bakeryClient = httpbakery.NewClient()
 	r.bakeryClient.Client = r.http
+	//r.bakeryClient.AddInteractor(httpbakery.WebBrowserInteractor{})
```
Comment (stgraber) [inline client/lxd.go]: leftover?

**Comment by stgraber** [comment]
I think the server keys should become:
 - candid.api.url
 - candid.domains

To match what we have for MAAS these days.

File: doc/server.md [hunk 4]
```diff
@@ -10,14 +10,14 @@ currently supported:
 
 Key                             | Type      | Default   | API extension            | Description
 :--                             | :---      | :------   | :------------            | :----------
+candid.api.url                  | string    | -         | macaroon\_authentication | URL of the the external authentication endpoint using Candid
```
Comment (stgraber) [inline doc/server.md]: Should get a new API extension for it

**Comment by monstermunchkin** [comment]
Done

**Comment by stgraber** [comment]
Looks good, will land soon.

**Comment by stgraber** [comment]
Needs quick rebase for i18n

**Comment by monstermunchkin** [comment]
Done

**Pull request merged**